src/binary/decimal.rs

```diff
@@ -11,14 +11,17 @@ use crate::binary::var_int::VarInt;
 use crate::binary::var_uint::VarUInt;
 use crate::ion_data::IonEq;
 use crate::result::IonResult;
+use crate::types::integer::UIntData;
 use crate::types::{Coefficient, Decimal, Sign, UInt};
 use crate::IonError;
 
 const DECIMAL_BUFFER_SIZE: usize = 32;
 const DECIMAL_POSITIVE_ZERO: Decimal = Decimal {
     coefficient: Coefficient {
         sign: Sign::Positive,
-        magnitude: UInt::U64(0),
+        magnitude: UInt {
+            data: UIntData::U64(0),
+        },
```

Comment:
🗺️ `magnitude: 064.into()` would be more concise, but the trait method `into()` is not `const`.